src/aggregator/aggregator/flush_mgr_options.go

```diff
@@ -39,7 +39,7 @@ const (
 )
 
 var (
-	defaultWorkerPoolSize = int(math.Max(float64(runtime.NumCPU()/8), 1.0))
+	defaultWorkerPoolSize = int(math.Max(float64(runtime.GOMAXPROCS(0)/8), 1.0))
```

Comment:
for my understanding... `runtime.NumCPU` reports the total number of cores on the VM (e.g 64) and `runtime.GOMAXPROCS` reports the number of cores visible to the container (e.g 4)?

Reply:
No, it's exactly the same unless you go the extra mile and set GOMAXPROCS for the container, which you'd usually do via container runtime / entrypoint script:

> GOMAXPROCS sets the maximum number of CPUs that can be executing simultaneously and returns the previous setting. It defaults to the value of runtime.NumCPU. If n < 1, it does not change the current setting. This call will go away when the scheduler improves.

There are packages that do this for you, by reading cgroup quota, example - https://pkg.go.dev/go.uber.org/automaxprocs